From 3029bc92f20d249ebea8779e46f6007aa0519b9a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=A9mie=20Galarneau?=
 <jeremie.galarneau@efficios.com>
Date: Thu, 19 Dec 2019 22:13:10 -0500
Subject: [PATCH lttng-tools] Fix: sessiond: check for lttng-modules ABI 2.1
 rather than 2.8
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The clear patchset introduces a regression that breaks session
rotations when the kernel domain is used. The 2.8 lttng-modules ABI
does not exist yet.

The packet sequence number functionality was introduced in LTTng 2.8,
which introduced the 2.1 kernel tracer ABI.

Signed-off-by: Jérémie Galarneau <jeremie.galarneau@efficios.com>
Change-Id: Ief97e7e25f6f0fcb5b5b97b39abb417c1eb327ec
---
 src/bin/lttng-sessiond/kernel.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/src/bin/lttng-sessiond/kernel.c
+++ b/src/bin/lttng-sessiond/kernel.c
@@ -1443,9 +1443,10 @@
 	}
 
 	/*
-	 * Packet sequence number was introduced in 2.8
+	 * Packet sequence number was introduced in LTTng 2.8,
+	 * lttng-modules ABI 2.1.
 	 */
-	if (abi.major >= 2 && abi.minor >= 8) {
+	if (abi.major >= 2 && abi.minor >= 1) {
 		/* Supported */
 		ret = 1;
 	} else {
